plugins/common/tls/config.go

```diff
@@ -206,23 +210,62 @@ func (c *ServerConfig) TLSConfig() (*tls.Config, error) {
 func makeCertPool(certFiles []string) (*x509.CertPool, error) {
 	pool := x509.NewCertPool()
 	for _, certFile := range certFiles {
-		pem, err := os.ReadFile(certFile)
+		cert, err := os.ReadFile(certFile)
 		if err != nil {
 			return nil, fmt.Errorf("could not read certificate %q: %w", certFile, err)
 		}
-		if !pool.AppendCertsFromPEM(pem) {
+		if !pool.AppendCertsFromPEM(cert) {
 			return nil, fmt.Errorf("could not parse any PEM certificates %q: %w", certFile, err)
 		}
 	}
 	return pool, nil
 }
 
-func loadCertificate(config *tls.Config, certFile, keyFile string) error {
-	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+func loadCertificate(config *tls.Config, certFile, keyFile, privateKeyPassphrase string) error {
+	var (
+		cert      tls.Certificate
+		certBytes []byte
+		keyBytes  []byte
+		err       error
+	)
+	certBytes, err = os.ReadFile(certFile)
 	if err != nil {
-		return fmt.Errorf("could not load keypair %s:%s: %w", certFile, keyFile, err)
+		return err
```

Comment:
Can we please add some context here to allow easier debugging
```suggestion
		return fmt.Errorf("could not load certificate %q: %w", certFile, err)
```